gprofiler/metadata/application_metadata.py

```diff
@@ -50,19 +50,41 @@ def get_metadata(self, process: Process) -> Optional[Dict]:
                 self._clear_cache()
             try:
                 metadata = self.make_application_metadata(process)
-            except NoSuchProcess:
+            except (NoSuchProcess, ZombieProcess):
                 # let our caller handler this
                 raise
             except Exception:
                 # log only the first _MAX_METADATA_EXCEPTION_LOGS exceptions; I expect the same exceptions to
                 # be repeated again and again, so it's enough to log just a handful of them.
                 if self._metadata_exception_logs_count < self._MAX_METADATA_EXCEPTION_LOGS:
-                    logger.exception(f"Exception while collecting metadata in {self.__class__.__name__}!")
+                    logger.exception(
+                        f"Exception while collecting metadata in {self.__class__.__name__}!", pid=process.pid
+                    )
                     self._metadata_exception_logs_count += 1
             else:
                 self._cache[process] = metadata
 
         return metadata
 
     def make_application_metadata(self, process: Process) -> Dict[str, Any]:
-        return {"exe": process.exe(), "execfn": None if is_windows() else read_process_execfn(process)}
+        md = {}
+
+        try:
+            exe = process_exe(process)
+        except (NoSuchProcess, ZombieProcess):
+            raise  # let caller handle
+        except Exception as e:
+            logger.exception("Exception while reading process exe", pid=process.pid)
+            exe = f"error: {e.__class__.__name__}"
+        md["exe"] = exe
+
+        try:
+            execfn = "error: not supported on Windows" if is_windows() else read_process_execfn(process)
+        except (NoSuchProcess, ZombieProcess):
+            raise  # let caller handle
+        except Exception as e:
+            logger.exception("Exception while reading process execfn", pid=process.pid)
+            execfn = f"error: {e.__class__.__name__}"
```

Comment:
Do we usually leave errors where the values should be instead of setting to None? It feels like a bad practice that may lead to other code parsing the error as if it's a real exe/execfn

Reply:
I want the error to be clear, as the type of error itself might be part of the logic. Since the error value is not a legitimate value and can be clearly distinguished, I think it's okay.
Alternatively, I can use `None` and add another `exe_error` and `execfn_error` fields. Makes more sense?

Reply:
If we have full control over the parsing side then it's ok I guess, I mainly wanted to bring some attention to this 👍